src/pages/index.js

```diff
@@ -33,61 +47,130 @@ document.addEventListener("DOMContentLoaded", () => {
   addFormValidator = new FormValidator(validationSettings, dom.cardAddForm);
   addFormValidator.enableValidation();
 
-  const cardSection = new Section(
-    {
-      items: initialCards,
-      renderer: renderCard,
-    },
-    selectors.cardSection
+  avatarFormValidator = new FormValidator(
+    validationSettings,
+    dom.avatarEditForm
   );
-  cardSection.renderItems();
+  avatarFormValidator.enableValidation();
 
-  function renderCard(item) {
-    const card = new Card(item, selectors.cardTemplate, handleImageClick);
-    const cardEl = card.getView();
-    cardSection.addItem(cardEl);
-  }
+  // Confirm Delete Modal
+  const confirmDeletePopup = new PopupWithConfirm({
+    popupSelector: "#confirm-delete-modal",
+  });
+  confirmDeletePopup.setEventListeners();
+
+  // Card Section
+  cardSection = new Section(
+    { items: [], renderer: renderCard },
+    selectors.cardSection
+  );
 
+  // Add Card Popup
   const cardAddPopupWithForm = new PopupWithForm(
     "#add-card-modal",
     (values) => {
-      return new Promise((resolve) => {
-        renderCard({
-          name: values.title,
-          link: values.url,
+      return api
+        .addCard({ name: values.title, link: values.url })
+        .then((newCard) => {
+          renderCard(newCard);
+          cardAddPopupWithForm.close(); // ← close on success
+          addFormValidator.disableSubmitButton();
+        })
+        .catch((err) => {
```

Comment:
You don't need to **catch**  errors  in `index.js` because you **catch** them inside `PopupWithForm` in a `.catch` block already

Please, delete it


**Please, correct it everywhere in the code**